projects/pendle/index.js

```diff
@@ -1,72 +1,55 @@
-    "0x8315BcBC2c5C1Ef09B71731ab3827b0808A2D6bD": "PendleAaveUSDCDec22Market",
-    "0x944d1727d0b656f497e74044ff589871c330334f": "PendleCompoundDAIDec21Market",
-    "0xB26C86330FC7F97533051F2F8cD0a90C2E82b5EE": "PendleCompoundDAIDec22Market",
-    "0x79c05Da47dC20ff9376B2f7DbF8ae0c994C3A0D0": "PendleSLP-ETHUSDC/USDC22Market",
-
-    "0x3483194Ac09097463CB426D8c0fc31c1476212f9": "PendleLpHolder_AaveUSDCDec21",
-    "0x76A16d9325E9519Ef1819A4e7d16B168956f325F": "PendleLpHolder_AaveUSDCDec22",
-    "0x5444070C9252BC6162A78fcFf66CF8Dcc3e729B8": "PendleLpHolder_CompoundDAIDec21",
-    "0x2F16B22C839FA995375602562ba5dD15A22d349d": "PendleLpHolder_CompoundDAIDec22",
-    "0xb69DA28b6B5DdF0fd4Fee4823A3Ffd2243A13C92": "PendleLpHolder_SLP-ETHUSDC/USDC22"
-})
-
-const SingleStaking = "0x07282F2CEEbD7a65451Fcd268b364300D9e6D7f5"
-
-const pool2contracts = Object.keys({
-    "0x685d32f394a5F03e78a1A0F6A91B4E2bf6F52cfE": "PendleSLP-PENDLEETH/PENDLE22Market",
-    "0xab30397316d06572968d068d16f1e611c46474e2": "PendleLpHolder_SLP-PENDLEETH/PENDLE22",
-    "0xbFD6b497dCa3e5D1fA4BbD52996d400980C29Eb7": "PendleSushiswapComplexYieldTokenHolder"
-})
+const tokens = contracts.tokens;
+const fundedContracts = Object.keys(contracts.funded);
+const stakingContracts = Object.keys(contracts.staking);
+const otTokens = Object.keys(contracts.otTokens);
+const pool2Contracts = Object.keys(contracts.pool2);
 
 async function tvl(timestamp, block) {
     const balances = {}
-    await sumTokensAndLPsSharedOwners(balances, [
-        [aUSDC, false],
-        [cDAI, false],
-        [USDC, false],
-        [SUSHI, false],
-        [COMP, false],
-    ], contracts, block)
 
+    await sumTokensAndLPsSharedOwners(balances, [
+        [tokens["USDC"], false],
+        [tokens["aUSDC"], false],
+        [tokens["cDAI"], false],
+        [tokens["SLP_ETHUSDC"], true],
+        [tokens["SLP_PENDLEETH"], true],
+        [tokens["SUSHI"], false],
+        [tokens["COMP"], false]
+    ], fundedContracts, block)
+    delete balances[tokens["PENDLE"]]
     return balances
 }
-
+async function staking(timestamp, block) {
+    const staking = {}
+    await sumTokensAndLPsSharedOwners(staking, [
+        [tokens["USDC"], false],
+        [tokens["PENDLE"], false],
+        [tokens["SLP_OT_aUSDC_21"], true],
+        [tokens["SLP_OT_aUSDC_22"], true],
+        [tokens["SLP_OT_cDAI_21"], true], 
+        [tokens["SLP_OT_cDAI_22"], true],
+        [tokens["SLP_OT_ETHUSDC_22"], true],
+        [tokens["SLP_OT_PEP_22"], true]
+    ], stakingContracts, block)
+    for (token of otTokens) {
+        delete staking[token.toLowerCase()]
+    }
+    return staking
+}
 async function pool2(timestamp, block) {
     const pool2 = {}
     await sumTokensAndLPsSharedOwners(pool2, [
-        [PENDLE, false],
-        [SUSHI, false],
-    ], pool2contracts, block)
+        [tokens["SLP_PENDLEETH"], true],
+        [tokens["PENDLE"], false],
```

Comment:
I think we made a mistake before, why were we counting PENDLE and SUSHI? Imo should have only been counting the SLP

Reply:
Part of the SLP staking rewards get paid in SUSHI